sdk/python/feast/infra/feature_servers/multicloud/offline/Dockerfile.builder.arrow

```diff
@@ -0,0 +1,131 @@
+    ARROW_PROTOBUF_URL="/tmp/arrow/cpp/arrow-thirdparty/protobuf-v21.3.tar.gz" \
+    ARROW_RAPIDJSON_URL="/tmp/arrow/cpp/arrow-thirdparty/rapidjson-232389d4f1012dddec4ef84861face2d2ba85709.tar.gz" \
+    ARROW_RE2_URL="/tmp/arrow/cpp/arrow-thirdparty/re2-2022-06-01.tar.gz" \
+    ARROW_S2N_TLS_URL="/tmp/arrow/cpp/arrow-thirdparty/s2n-v1.3.35.tar.gz" \
+    ARROW_SUBSTRAIT_URL="/tmp/arrow/cpp/arrow-thirdparty/substrait-0.44.0.tar.gz" \
+    ARROW_SNAPPY_URL="/tmp/arrow/cpp/arrow-thirdparty/snappy-1.1.10.tar.gz" \
+    ARROW_THRIFT_URL="/tmp/arrow/cpp/arrow-thirdparty/thrift-0.16.0.tar.gz" \
+    ARROW_UCX_URL="/tmp/arrow/cpp/arrow-thirdparty/ucx-1.12.1.tar.gz" \
+    ARROW_UTF8PROC_URL="/tmp/arrow/cpp/arrow-thirdparty/utf8proc-v2.7.0.tar.gz" \
+    ARROW_XSIMD_URL="/tmp/arrow/cpp/arrow-thirdparty/xsimd-13.0.0.tar.gz" \
+    ARROW_ZLIB_URL="/tmp/arrow/cpp/arrow-thirdparty/zlib-1.3.1.tar.gz" \
+    ARROW_ZSTD_URL="/tmp/arrow/cpp/arrow-thirdparty/zstd-1.5.6.tar.gz" \
+    && \
+    cmake \
+    -DCMAKE_INSTALL_PREFIX=$ARROW_HOME \
+    -DARROW_COMPUTE=ON \
+    -DARROW_ACERO=ON \
+    -DARROW_WITH_BROTLI=ON \
+    -DARROW_WITH_BZ2=ON \
+    -DARROW_CSV=ON \
+    -DARROW_DATASET=ON \
+    -DARROW_FILESYSTEM=ON \
+    -DARROW_FLIGHT=ON \
+    -DARROW_FLIGHT_SQL=ON \
+    -DARROW_GANDIVA=ON \
+    -DARROW_GCS=ON \
+    -DARROW_HDFS=ON \
+    -DARROW_JEMALLOC=ON \
+    -DARROW_JSON=ON \
+    -DARROW_WITH_LZ4=ON \
+    -DARROW_MIMALLOC=ON \
+    -DARROW_PARQUET=ON \
+    -DARROW_S3=ON \
+    -DARROW_WITH_SNAPPY=ON \
+    -DARROW_SUBSTRAIT=ON \
+    -DARROW_WITH_RE2=ON \
+    -DARROW_WITH_UTF8PROC=ON \
+    -DARROW_TENSORFLOW=ON \
+    -DARROW_WITH_ZLIB=ON \
+    -DARROW_WITH_ZSTD=ON \
+    -DARROW_BUILD_SHARED=ON \
+    -S /tmp/arrow/cpp \
+    -B ${APP_ROOT}/src/arrow-build && \
+    \
+    cmake --build ${APP_ROOT}/src/arrow-build --target install && \
+    \
+    source /tmp/cachi2.env && \
+    pip install --no-binary :all: -r /tmp/offline/pyarrow17-wheel-build-requirements.txt && \
+    \
+    cd ${APP_ROOT}/src/arrow/python && \
+    PYARROW_PARALLEL=4 python setup.py build_ext --bundle-arrow-cpp bdist_wheel && \
+    pip install dist/pyarrow-*.whl && \
+    \
+    cd ${APP_ROOT}/src && \
+    rm -rf ${APP_ROOT}/src/arrow-build ${APP_ROOT}/src/arrow ${APP_ROOT}/arrow-dist
+
+# a higher numpy was required for the pyarrow wheel build, but the pyarrow module itself can run w/ a lesser version, "numpy>=1.16.6".
+# feast requires "numpy<2", so here we install numpy 1.x, as well as some other packages which will be needed for the feast build.
+RUN source /tmp/cachi2.env && \
+    pip install --no-binary :all: \
+    "numpy<2" \
+    # needed for ibis build
+    typing-extensions \
+    # required deps for other pip installs
+    aiohappyeyeballs \
+    aiosignal \
+    anyio \
+    attrs \
+    cffi \
+    frozenlist \ 
+    idna \
+    multidict \
+    propcache \
+    pycparser \
+    sniffio \
+    tqdm \
+    yarl
+
+RUN pip list
```

Comment:
If this is added for verifying pyarrow installation, can we do like 
`RUN python -c "import pyarrow; print(pyarrow.__version__)"` ?